openstack/clientconfig/requests.go

```diff
@@ -56,16 +57,26 @@ func LoadYAML() (map[string]Cloud, error) {
 		return nil, fmt.Errorf("failed to unmarshal yaml: %v", err)
 	}
 
+
 	return clouds.Clouds, nil
 }
 
-// GetCloudFromYAML will return a cloud entry from a clouds.yaml file.
+// GetCloudFromYAML will return a cloud entry from a the yaml files used to specify the Clouds.
 func GetCloudFromYAML(opts *ClientOpts) (*Cloud, error) {
-	clouds, err := LoadYAML()
+
+	privateClouds, err := LoadYAML("clouds.yaml")
```

Comment:
I think it makes most sense to handle the different files and merge the clouds in this function even if that means changing the signature of LoadYAML to take a filename as well.